pkg/controller/certificates/rootcacertpublisher/publisher.go

```diff
@@ -179,17 +203,22 @@ func (c *Publisher) processNextWorkItem() bool {
 	return true
 }
 
-func (c *Publisher) syncNamespace(ns string) (err error) {
+func (c *Publisher) syncNamespace(key string) (err error) {
 	startTime := time.Now()
 	defer func() {
 		recordMetrics(startTime, err)
-		klog.V(4).Infof("Finished syncing namespace %q (%v)", ns, time.Since(startTime))
+		klog.V(4).Infof("Finished syncing namespace %q (%v)", key, time.Since(startTime))
 	}()
 
-	cm, err := c.cmLister.ConfigMaps(ns).Get(RootCACertConfigMapName)
+	// Get the clusterName and name from the key.
+	_, clusterAwareName, err := cache.SplitMetaNamespaceKey(key)
+	clusterName, name := clusters.SplitClusterAwareKey(clusterAwareName)
+
+	cm, err := c.client.CoreV1().ConfigMaps(name).Get(genericapirequest.WithCluster(context.TODO(), genericapirequest.Cluster{Name: clusterName}), RootCACertConfigMapName, metav1.GetOptions{})
```

Comment:
this we cannot do, we have to use the lister

Reply:
thanks! fixed.